domain-model/src/main/java/uk/gov/hmcts/cmc/domain/constraints/ValidFullAdmissionConstraintValidator.java

```diff
@@ -0,0 +1,103 @@
+
+    @Override
+    public boolean isValid(FullAdmissionResponse fullAdmissionResponse, ConstraintValidatorContext context) {
+        switch (fullAdmissionResponse.getPaymentOption()) {
+            case IMMEDIATELY:
+                return validateImmediately(fullAdmissionResponse, context);
+            case FULL_BY_SPECIFIED_DATE:
+                return validateBySetDate(fullAdmissionResponse, context);
+            case INSTALMENTS:
+                return validateInstalments(fullAdmissionResponse, context);
+            default:
+                return false;
+        }
+    }
+
+    private boolean validateImmediately(FullAdmissionResponse value, ConstraintValidatorContext context) {
+        boolean valid = true;
+        String immediately = IMMEDIATELY.getDescription();
+
+        if (value.getPaymentDate().isPresent()) {
+            setValidationErrors(context, Fields.PAYMENT_DATE, mayNotBeProvidedError("paymentType", immediately));
+            valid = false;
+        }
+
+        if (value.getRepaymentPlan().isPresent()) {
+            setValidationErrors(context, Fields.REPAYMENT_PLAN, mayNotBeProvidedError("paymentType", immediately));
+            valid = false;
+        }
+
+        if (value.getStatementOfMeans().isPresent()) {
+            setValidationErrors(context, Fields.STATEMENT_OF_MEANS, mayNotBeProvidedError("paymentType", immediately));
+            valid = false;
+        }
+
+        return valid;
+    }
+
+    private boolean validateBySetDate(FullAdmissionResponse value, ConstraintValidatorContext context) {
+        boolean valid = true;
+        String bySetDate = FULL_BY_SPECIFIED_DATE.getDescription();
+
+        if (!value.getPaymentDate().isPresent()) {
+            setValidationErrors(context, Fields.PAYMENT_DATE, mayNotBeNullError("paymentType", bySetDate));
+            valid = false;
+        }
+
+        if (value.getRepaymentPlan().isPresent()) {
+            setValidationErrors(context, Fields.REPAYMENT_PLAN, mayNotBeProvidedError("paymentType", bySetDate));
+            valid = false;
+        }
+
+        if (!value.getStatementOfMeans().isPresent()) {
+            setValidationErrors(context, Fields.STATEMENT_OF_MEANS, mayNotBeNullError("paymentType", bySetDate));
+            valid = false;
+        }
+
+        return valid;
+    }
+
+    private boolean validateInstalments(FullAdmissionResponse value, ConstraintValidatorContext context) {
+        boolean valid = true;
+        String instalments = INSTALMENTS.getDescription();
+
+        if (value.getPaymentDate().isPresent()) {
+            setValidationErrors(context, Fields.PAYMENT_DATE, mayNotBeProvidedError("paymentType", instalments));
+            valid = false;
+        }
+
+        if (!value.getRepaymentPlan().isPresent()) {
+            setValidationErrors(context, Fields.REPAYMENT_PLAN, mayNotBeNullError("paymentType", instalments));
+            valid = false;
+        }
+
+        if (!value.getStatementOfMeans().isPresent()) {
```

Comment:
We cannot have it here since SoM is only for individuals and sole traders.

Reply:
removed